tests/phpunit/tests/option/option.php

```diff
@@ -312,4 +312,214 @@ public function data_option_autoloading() {
 			array( 'autoload_false', false, 'no' ),
 		);
 	}
+
+	/**
+	 * Test prime_options function.
+	 *
+	 * @ticket 58962
+	 *
+	 * @covers ::prime_options
+	 */
+	public function test_prime_options() {
+		// Create some options to prime.
+		$options_to_prime = array(
+			'option1',
+			'option2',
+			'option3',
+		);
+
+		// Make sure options are not in cache or database initially.
+		$this->assertFalse( wp_cache_get( 'option1', 'options' ) );
+		$this->assertFalse( wp_cache_get( 'option2', 'options' ) );
+		$this->assertFalse( wp_cache_get( 'option3', 'options' ) );
+
+		// Call the prime_options function to prime the options.
+		prime_options( $options_to_prime );
+
+		// Check that options are now in the cache.
+		$this->assertEquals( get_option( 'option1' ), wp_cache_get( 'option1', 'options' ) );
+		$this->assertEquals( get_option( 'option2' ), wp_cache_get( 'option2', 'options' ) );
+		$this->assertEquals( get_option( 'option3' ), wp_cache_get( 'option3', 'options' ) );
```

Comment:
This logic looks incorrect: None of these options actually exists, so none of them is actually set in the cache; the logic here only works because the default return value for non-existent options is `false`, the same as for cache misses.

You'll need to first set values for these options (with `update_option()`), then delete their cache, and then run this logic.

Reply:
Make sense. Initially i don't think about it. Thanks for catching it. I have address it https://github.com/WordPress/wordpress-develop/pull/4970/commits/b191e7b8d719a92c0e1f192c3360333ce5edc340